src/VisualStudio/Core/Def/ProjectSystem/VisualStudioProject.BatchingDocumentCollection.cs

```diff
@@ -473,18 +479,9 @@ public void ProcessDynamicFileChange(string projectSystemFilePath, string worksp
                                 _project.Id, _project._filePath, projectSystemFilePath, CancellationToken.None).WaitAndGetResult_CanCallOnBackground(CancellationToken.None);
 
                             // Right now we're only supporting dynamic files as actual source files, so it's OK to call GetDocument here
-                            var document = w.CurrentSolution.GetRequiredDocument(documentId);
-
-                            var documentInfo = DocumentInfo.Create(
-                                document.Id,
-                                document.Name,
-                                document.Folders,
-                                document.SourceCodeKind,
-                                loader: fileInfo.TextLoader,
-                                document.FilePath,
-                                document.State.Attributes.IsGenerated,
-                                document.State.Attributes.DesignTimeOnly,
-                                documentServiceProvider: fileInfo.DocumentServiceProvider);
+                            var attributes = w.CurrentSolution.GetRequiredDocument(documentId).State.Attributes;
+
+                            var documentInfo = new DocumentInfo(attributes, fileInfo.TextLoader, new LoadTextOptions(_project.ChecksumAlgorithm), fileInfo.DocumentServiceProvider);
```

Comment:
A case where I'm confused why we need to pass the load options, since this should just inherit from the project.